ocean_lib/services/service.py

```diff
@@ -272,3 +299,73 @@ def update_compute_values(
         ] = trusted_algo_publishers
         self.compute_values["allowNetworkAccess"] = allow_network_access
         self.compute_values["allowRawAlgorithm"] = allow_raw_algorithm
+
+
+class ConsumerParameters:
+    def __init__(
+        self,
+        name: str,
+        type: Optional[str] = None,
+        label: Optional[str] = None,
+        required: Optional[bool] = False,
+        default: Optional[str] = None,
+        description: Optional[str] = None,
+        options: Optional[List[str]] = None,
+    ) -> None:
+        self.name = name
+        self.type = type
+        self.label = label
+        self.required = required
+        self.default = default
+        self.description = description
+
+        if options is not None and not isinstance(options, list):
+            raise TypeError("Options should be a list")
```

Comment:
Unfortunately that's why I had to add the Type Error, because I can't add `@enforce_types` due to errors. It's the same old problem we had previously. So I manually validated it.

Reply:
I see.  Okay.